lib/security/secureboot.pm

```diff
@@ -0,0 +1,44 @@
+# secure boot libs
+#
+# Copyright SUSE LLC
+# SPDX-License-Identifier: FSFAP
+# Summary: secure boot enablment/disablement support
+#
+# Maintainer: QE Security <none@suse.de>
+
+package security::secureboot;
+
+use base "opensusebasetest";
+use strict;
+use warnings;
+use testapi;
+use Exporter 'import';
+use Utils::Architectures 'is_aarch64';
+use bootloader_setup qw(tianocore_disable_secureboot);
+use version_utils qw(is_sle);
+
+our @EXPORT = qw(handle_secureboot);
+
+sub handle_secureboot {
+    my ($self, $sb_state, $sb_opt) = @_;
+
+    my $boot_method = ((is_aarch64 && is_sle('>=16')) ? 'wait_boot_past_bootloader' : 'wait_boot');
+
+    if ($sb_state =~ /disabled/) {
+        record_soft_failure('poo#179960 - Skip re-enabling/disabling secure boot – it was initially disabled.');
+        $self->$boot_method(textmode => 1);
+        return;
+    }
+
+    record_info('bsc#1189988:', 'Disabling Secure Boot due to known IMA fix mode issue');
+    $self->wait_grub(bootloader_time => 200);
+
+    if (defined $sb_opt && $sb_opt eq 're_enable') {
```

Comment:
`$self->tianocore_disable_secureboot($sb_opt eq 're_enable' ? 're_enable' : ());`

Reply:
`defined $sb_opt &&` could probably be left out as well, but we might get warnings, albeit not fatal errors.

Reply:
I find it less clear and readable in this way to be honest.